### PR TITLE
Add missing return types to interfaces

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/issues/47551#issuecomment-1429726162
| License       | MIT
| Doc PR        | N/A

This PR adds missing return type documentation to interfaces listed in https://github.com/symfony/symfony/issues/47551#issuecomment-1429726162. I've started at the top and processed bundles + components starting with letters A-E.

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php [hunk 1]
```diff
@@ -705,7 +711,10 @@ private function createHashers(array $hashers, ContainerBuilder $container)
         ;
     }
 
-    private function createHasher(array $config)
+    /**
+     * @return Reference|array{class: class-string, arguments: list<mixed>}
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: not only class+arguments - not sure it's worth listing them all?

**Comment by nicolas-grekas** [comment]
Please keep `https://github.com/symfony/symfony/blob/6.3/.github/expected-missing-return-types.diff` is sync while working on this, it allows cascading this to the all other classes.

**Review by wouterj — APPROVED** [review]

**Comment by derrabus** [comment]
> Please keep `https://github.com/symfony/symfony/blob/6.3/.github/expected-missing-return-types.diff` is sync

Is there an easy way to do this?

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: Right, I missed the spot where the array is returned as-is. I'll change it to `array<string, mixed>`.

**Comment by stof** [comment]
@derrabus reading the instructions at the top of the file is a good start: https://github.com/symfony/symfony/blob/6.3/.github/expected-missing-return-types.diff#L1

**Comment by derrabus** [comment]
> @derrabus reading the instructions at the top of the file is a good start: https://github.com/symfony/symfony/blob/6.3/.github/expected-missing-return-types.diff#L1

Those commands were not fully compatible with the macOS flavor of `sed` it seems…

> sed: 1: "composer.json": command c expects \ followed by text
> …
> sed: 1: "/^diff /Q": invalid command code Q

I've committed the resulting file. Not sure if it was updated correctly. 😇

File: .github/expected-missing-return-types.diff [hunk 2]
```diff
@@ -3381,13 +3666,13 @@ index cd8ebfe0f7..36cd2b93a4 100644
          $parameters = $container->getParameterBag()->all();
 @@ -169,10 +169,10 @@ class MergeExtensionConfigurationContainerBuilder extends ContainerBuilder
      }
-
```
Comment (nicolas-grekas) [inline .github/expected-missing-return-types.diff]: these should be preserved :)

**Comment by nicolas-grekas** [comment]
Don't miss looking at the failing tests, they're 100% related :)

**Comment by derrabus** [comment]
> Don't miss looking at the failing tests, they're 100% related :)

The notifier test failures about static methods becoming non-static don't look related, but the rest are. Looking into them.

**Comment by derrabus** [comment]
btw: fabbot wants me to remove `@return mixed` annotations. I think it should stop doing that.

**Comment by nicolas-grekas** [comment]
Can you please rebase to latest 6.3 to see if this is going to fix Notifier?
Please also always sync the expected missing type when changing any return type.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**